gson-problem/src/test/java/org/zalando/problem/GsonProblemTest.java

```diff
@@ -0,0 +1,272 @@
+package org.zalando.problem;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.stream.JsonReader;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.*;
+import java.net.URI;
+import java.net.URL;
+import java.util.List;
+import java.util.Objects;
+
+import static com.jayway.jsonassert.JsonAssert.with;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.hobsoft.hamcrest.compose.ComposeMatchers.hasFeature;
+import static org.zalando.problem.Status.BAD_REQUEST;
+
+class GsonProblemTest {
+
+    private static Gson gson;
+
+    @BeforeAll
+    static void init() {
+        GsonBuilder builder = new GsonBuilder();
+        ProblemAdapterFactory problemAdapter = new ProblemAdapterFactory(false);
+        // Register known sub-types
+        problemAdapter
+                .registerSubtype(InsufficientFundsProblem.class, InsufficientFundsProblem.TYPE)
+                .registerSubtype(OutOfStockException.class, OutOfStockException.TYPE);
+        builder.registerTypeAdapterFactory(problemAdapter);
+        gson = builder.create();
+    }
+
+    private static JsonReader getReader(final String name) throws FileNotFoundException {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        URL resource = Objects.requireNonNull(loader.getResource(name), () -> "resource " + name + " not found.");
+        return new JsonReader(new FileReader(new File(resource.getPath())));
+    }
+
+    private String getStackTrace(final Throwable throwable) {
+        final StringWriter writer = new StringWriter();
+        throwable.printStackTrace(new PrintWriter(writer));
+        return writer.toString();
+    }
+
+    @Test
+    void shouldSerializeDefaultProblem() {
```

Comment:
There is no serialization test that takes a custom problem, e.g. `InsufficientFundsProblem` and verifies that custom properties are serialized correctly.

Reply:
This would need some time. Will fix asap.